File: src/led/json/interpreter/strategies/V0_0_1InterpreterStrategy.java
package led.json.interpreter.strategies;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Stream;

import com.google.gson.Gson;

import led.ColorRGB;
import led.json.LEDJsonProcedure;
import led.json.LEDStateArray;

public class V0_0_1InterpreterStrategy implements IInterpreterStrategy {

	@Override
	public ColorRGB[] interpretLine(String _line) {
		String[] colorData = _line.split(" ");
		
		ArrayList<ColorRGB> colors = new ArrayList<ColorRGB>();
		for (int i = 0; i < colorData.length; i++) {
			String[] ledData = colorData[i].split("-");
			
			int[] arr = Stream.of(ledData).mapToInt(Integer::parseInt).toArray();
			
			colors.add(new ColorRGB(arr[0], arr[1], arr[2]));
		}
		
		return colors.toArray(new ColorRGB[colors.size()]);
	}

	@Override
	public LEDJsonProcedure interpretJson(String _json) {
		LEDJsonProcedure001Uncompiled jsonProc = new Gson().fromJson(_json, LEDJsonProcedure001Uncompiled.class);
		LEDJsonProcedure result = new LEDJsonProcedure();
		result.mMetaInfo = jsonProc.mMetaInfo;
		List<LEDStateArray> l = new ArrayList<LEDStateArray>();
		for (int i = 0; i < 300; i++) {
			l.add(new LEDStateArray());
			l.get(i).mLEDState = interpretLine(jsonProc.mLEDStates[i]);
		}
		result.mLEDStates = new LEDStateArray[300];
		l.toArray(result.mLEDStates);
		System.out.println(result.mLEDStates.length);
		return result;
	}

}

File: src/led/json/interpreter/strategies/V0_0_1InterpreterStrategy.kt
package led.json.interpreter.strategies

import led.ColorRGB
import led.json.LEDJsonProcedure
import com.google.gson.annotations.SerializedName
import led.json.MetaInfo
import led.json.interpreter.strategies.IInterpreterStrategy
import com.google.gson.Gson
import java.util.function.ToIntFunction
import led.json.interpreter.strategies.LEDJsonProcedure001Uncompiled
import led.json.LEDStateArray
import java.util.ArrayList
import java.util.stream.Stream

class V0_0_1InterpreterStrategy : IInterpreterStrategy {
    override fun interpretLine(_line: String?): Array<ColorRGB> {
        val colorData = _line!!.split(" ".toRegex()).toTypedArray()
        val colors = ArrayList<ColorRGB>()
        for (i in colorData.indices) {
            val ledData = colorData[i].split("-".toRegex()).toTypedArray()
            val arr = Stream.of(*ledData).mapToInt { s: String -> s.toInt() }.toArray()
            colors.add(ColorRGB(arr[0], arr[1], arr[2]))
        }
        return colors.toTypedArray()
    }

    override fun interpretJson(_json: String?): LEDJsonProcedure? {
        val jsonProc = Gson().fromJson(_json, LEDJsonProcedure001Uncompiled::class.java)
        val result = LEDJsonProcedure()
        result.mMetaInfo = jsonProc.mMetaInfo
        val l: MutableList<LEDStateArray> = ArrayList()
        for (i in 0..299) {
            l.add(LEDStateArray())
            l[i].mLEDState = interpretLine(jsonProc.mLEDStates[i])
        }
        result.mLEDStates = arrayOfNulls(300)
        println(result.mLEDStates.size)
        return result
    }
}